Server/Server.Exec/ArgParser.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <argparse/argparse.hpp>
+#include <FileLogger.hpp>
+
+/**
+ * @class ArgParser.
+ * @brief This class parses arguments with which application is started.
+ * @details
+ * For example: "--port 5432 --dbport 65001 --dbname juniorgram --hostaddr 127.0.0.1 --user postgres --password postgres".
+ * Sending the port to the Server class constructor and list of all arguments to the PostgreAdapter Instance Method.
+ */
+
+class ArgParser
+{
+public:
+    /**
+     *@brief It's deleted default constructor. There is no need to use this.
+     */
+    ArgParser() = delete;
+   
+    /**
+     *@brief It's deleted copy constructor. There is no need to use this.
+     */
+    ArgParser(const ArgParser&) = delete;
+   
+    /**
+     * @brief Used to take parameters from main function.
+     * @details Parses arguments which were set when application was run.
+     * @param int argc(amount of arguments), const char** argv(C-string arguments).
+     */
+    ArgParser(int argc, const char** argv) 
+    {
+        _program = argparse::ArgumentParser("ServerArguments", "0.6", argparse::default_arguments::none);
+        
+        _program.add_description("Server side of juniorgram messenger. Just have a fun!");
+        _program.add_epilog("Some part of program could be gluchennie.");
+
+        _program.add_argument("--port")
+            .default_value("5432")
+            .help("Set the port value for server needs")
+            .action([](std::string value) -> int {
+                int port = std::stoi(value);
```

Comment:
I see restrictions only for positive numbers, but what if we have number below zero? Maybe I've lost check about this. Also, why int. I mean, everywhere when we don't need to see numbers below zero we use unsigned version and sometimes it says something about restrictions and properies of variable (like I said about that we don't look forward to seeing negative numbers)

Reply:
> I see restrictions only for positive numbers, but what if we have number below zero? Maybe I've lost check about this. Also, why int. I mean, everywhere when we don't need to see numbers below zero we use unsigned version and sometimes it says something about restrictions and properies of variable (like I said about that we don't look forward to seeing negative numbers)

We need to branch for this situation, so that in case of a negative number the builder will not start or it will start on one of the default ports 65001 or 5432.